sources/corepackUtils.ts

```diff
@@ -85,6 +85,7 @@ export async function findInstalledVersion(installTarget: string, descriptor: De
     // If the dirname correspond to an in-range version and is not lower than
     // the previous best match (or if there is not yet a previous best match),
     // it's our new best match.
+    // @ts-expect-error TODO: decipher why this produces an error
     if (range.test(name) && maxSV?.compare(name) !== 1) {
```

Comment:
No idea why this code generates the following error:

```
error TS2339: Property 'compare' does not exist on type 'never'.

88     if (range.test(name) && maxSV?.compare(name) !== 1) {
```

Reply:
```suggestion
    if (range.test(name) && maxSV && maxSV.compare(name) !== 1) {
```
This works so potentially a TypeScript bug?

Reply:
```suggestion
    if (range.test(name) && (maxSV as semver.SemVer | undefined)?.compare(name) !== 1) {
```
This also don't trigger the error, so definitely a TS bug, I guess we just have to wait for them to fix it.